app/components/ui/Navigation/DropdownSubmenu.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import navStyles from "./DesktopNavigation.module.scss";
+import categoryStyles from "./CategoryDropdown.module.scss";
+
+// This component is the dropdown submenu used by both DesktopMenuItems and CategoryDropdown, created in service of DRY code
+
+const DropdownSubmenu = ({
+  title,
+  links,
+  activeSubMenu,
+  handleMenuToggle,
+  uniqueKey,
+  menuRef,
+  variant = "navigation",
+}: {
+  title: string;
+  links: { id: number | string; url: string; text: string }[];
+  activeSubMenu: string | null;
+  handleMenuToggle: (key: string | null) => void;
+  uniqueKey: string;
+  menuRef: React.RefObject<HTMLDivElement>;
+  variant?: "navigation" | "category";
```

Comment:
I feel like you're really close to an interface that is entirely decoupled from implementation -- needing to declare a variant type -- that would be generic for any reusable dropdown case. Could you just pass the `styles` parameter from the outside component rather than using the enum as a switch? 

```
styles?: {
    [key: string]: string;
};
```